toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoComposeState.kt

```diff
@@ -17,11 +17,69 @@
 
 package com.arcgismaps.toolkit.geocompose
 
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.ViewpointType
+import kotlinx.coroutines.channels.Channel
+
 /**
  * Represents the state for the GeoCompose.
  *
  * @since 200.3.0
  */
 public sealed class GeoComposeState {
-    // TODO add implementation
+
+    internal val viewpointChannel = Channel<ViewpointOperation>()
+    internal val getCurrentViewpointChannel = Channel<GetCurrentViewpointOperation>()
+    internal val setViewpointChannel = Channel<SetViewpointOperation>()
+
+    /**
+     * Change the GeoCompose to the new [viewpoint] with animation.
+     * This function uses the standard animation duration.
+     *
+     * @since 200.3.0
+     */
+    public suspend fun setViewpointAnimated(
+        viewpoint: Viewpoint
+    ): Result<Boolean> =
+        ViewpointOperation.ViewpointAnimated(viewpoint).let {
+            viewpointChannel.send(it)
+            it.await()
+        }
+
+    /**
+     * Change the GeoCompose to the new [viewpoint] with animation, taking the given number
+     * of seconds to complete the navigation.
+     *
+     * @since 200.3.0
+     */
+    public suspend fun setViewpointAnimated(
+        viewpoint: Viewpoint,
+        durationSeconds: Float
+    ): Result<Boolean> =
+        ViewpointOperation.ViewpointAnimated(viewpoint, durationSeconds).let {
+            viewpointChannel.send(it)
+            it.await()
+        }
+
+    /**
+     * Retrieve the current viewpoint for this Map using the given [viewpointType].
+     *
+     * @since 200.3.0
+     */
+    public suspend fun getCurrentViewpoint(viewpointType: ViewpointType): Result<Viewpoint> =
+        GetCurrentViewpointOperation.GetCurrentViewpoint(viewpointType).let {
+            getCurrentViewpointChannel.send(it)
+            it.await()
+        }
+
+    /**
+     * Change the Map to the new [viewpoint]. The viewpoint is updated instantaneously.
+     *
+     * @since 200.3.0
+     */
+    public suspend fun setViewpoint(viewpoint: Viewpoint) {
```

Comment:
Just realising that all our suspend functions need to return `Result<>` in order to propagate any errors. That means you need to change the function return type and wait on the operation, in case an error occurs while executing the function on the GeoView:
```suggestion
    public suspend fun setViewpoint(viewpoint: Viewpoint): Result<Unit> {
```